lib/jsonapi/matchers/attributes_included.rb

```diff
@@ -42,6 +45,35 @@ def matches?(target)
       def failure_message
         @failure_message || "expected attribute '#{@attribute_name}' to be included in #{@target.as_json.ai}"
       end
+
+      private
+
+      def value_exists?
+        if @expected_value.to_s == @value.to_s
+          true
+        else
+          @failure_message = "expected '#{@expected_value}' for key '#{@attribute_name}', but got '#{@value}'"
+          false
+        end
+      end
+
+      def record_exists?
+        data = @value.try(:[], 'data')
+
+        if data.is_a?(Array)
+          if data.map{|d| d['id']}.include?(@expected_record_id)
+            return true
+          else
+            @failure_message = "expected '#{@expected_record_id}' to be the an id in relationship '#{@attribute_name}', but got '#{@value['data']}'"
+            return false
+          end
+        elsif @expected_record_id == @value.try(:[], 'data').try(:[], 'id')
```

Comment:
#non-blocking

Have you played around with Ruby's safe navigation operator?
https://docs.ruby-lang.org/en/2.4.0/NEWS-2_3_0.html

Reply:
@4justinstewart yes! I saw that from the rails project and love it, but I wasn't sure if we wanted to leave this backwards compatible or not. will hold off for now. thank you for mentioning it!